com.unity.netcode.gameobjects/Prototyping/NetworkTransform.cs

```diff
@@ -152,25 +207,93 @@ public void NetworkSerialize(NetworkSerializer serializer)
+                ScaleYInterpolator = new TScale();
+                ScaleZInterpolator = new TScale();
+            }
+            else
+            {
+                PositionXInterpolator = new NoInterpolator<float>();
+                PositionYInterpolator = new NoInterpolator<float>();
+                PositionZInterpolator = new NoInterpolator<float>();
+                RotationInterpolator = new NoInterpolator<Quaternion>();
+                ScaleXInterpolator = new NoInterpolator<float>();
+                ScaleYInterpolator = new NoInterpolator<float>();
+                ScaleZInterpolator = new NoInterpolator<float>();
+            }
+
+            m_AllFloatInterpolators.Add(PositionXInterpolator);
+            m_AllFloatInterpolators.Add(PositionYInterpolator);
+            m_AllFloatInterpolators.Add(PositionZInterpolator);
+            m_AllFloatInterpolators.Add(ScaleXInterpolator);
+            m_AllFloatInterpolators.Add(ScaleYInterpolator);
+            m_AllFloatInterpolators.Add(ScaleZInterpolator);
+        }
+
+        private readonly List<IInterpolator<float>> m_AllFloatInterpolators = new List<IInterpolator<float>>(6);
+
         private Transform m_Transform; // cache the transform component to reduce unnecessary bounce between managed and native
-        internal NetworkState LocalNetworkState;
+
         internal readonly NetworkVariable<NetworkState> ReplNetworkState = new NetworkVariable<NetworkState>(new NetworkState());
         internal NetworkState PrevNetworkState;
+        internal NetworkState LocalAuthoritativeNetworkState;
+
+        public void ResetCurrentInterpolatedState()
+        {
+            PositionXInterpolator.ResetTo(ReplNetworkState.Value.PositionX);
+            PositionYInterpolator.ResetTo(ReplNetworkState.Value.PositionY);
+            PositionZInterpolator.ResetTo(ReplNetworkState.Value.PositionZ);
+
+            RotationInterpolator.ResetTo(Quaternion.Euler(ReplNetworkState.Value.Rotation));
+
+            ScaleXInterpolator.ResetTo(ReplNetworkState.Value.ScaleX);
+            ScaleYInterpolator.ResetTo(ReplNetworkState.Value.ScaleY);
+            ScaleZInterpolator.ResetTo(ReplNetworkState.Value.ScaleZ);
+        }
 
         // updates `NetworkState` properties if they need to and returns a `bool` indicating whether or not there was any changes made
         // returned boolean would be useful to change encapsulating `NetworkVariable<NetworkState>`'s dirty state, e.g. ReplNetworkState.SetDirty(isDirty);
-        internal bool UpdateNetworkState(ref NetworkState networkState)
+        internal bool UpdateNetworkStateCheckDirty(ref NetworkState networkState, double dirtyTime)
+        {
+            return UpdateNetworkStateCheckDirtyWithInfo(ref networkState, dirtyTime).isDirty;
+        }
+
+        private (bool isDirty, bool isPositionDirty, bool isRotationDirty, bool isScaleDirty) UpdateNetworkStateCheckDirtyWithInfo(ref NetworkState networkState, double dirtyTime)
```

Comment:
why?

Reply:
isRotationDirty is used elsewhere to know whether to produce the "non authoritative change" warning or not. Since rotation can mess with axis syncing, we're not doing that authority check with rotation

Reply:
did you see my other comment below related to this?

Reply:
as discussed, keeping custom isDirty check